src/sentry/static/sentry/app/views/performance/realUserMonitoring/index.tsx

```diff
@@ -0,0 +1,177 @@
+  projects: Project[];
+  selection: GlobalSelection;
+} & Pick<WithRouterProps, 'router'>;
+
+type State = {
+  eventView: EventView | undefined;
+};
+
+class RealUserMonitoring extends React.Component<Props> {
+  state: State = {
+    eventView: generateRumEventView(
+      this.props.location,
+      getTransactionName(this.props.location)
+    ),
+  };
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State): State {
+    return {
+      ...prevState,
+      eventView: generateRumEventView(
+        nextProps.location,
+        getTransactionName(nextProps.location)
+      ),
+    };
+  }
+
+  getDocumentTitle(): string {
+    const name = getTransactionName(this.props.location);
+
+    const hasTransactionName = typeof name === 'string' && String(name).trim().length > 0;
+
+    if (hasTransactionName) {
+      return [String(name).trim(), t('RUM')].join(' - ');
+    }
+
+    return [t('Summary'), t('RUM')].join(' - ');
+  }
+
+  renderNoAccess = () => {
+    const {router, organization, location} = this.props;
+
+    if (organization.features.includes('performance-view')) {
+      const transactionName = getTransactionName(this.props.location);
+      if (!transactionName) {
+        // If there is no transaction name, redirect to the Performance landing page
+
+        browserHistory.replace({
+          pathname: `/organizations/${organization.slug}/performance/`,
+          query: {
+            ...location.query,
+          },
+        });
+        return null;
+      }
+
+      return (
+        <Redirect
+          router={router}
+          to={transactionSummaryRouteWithQuery({
+            orgSlug: organization.slug,
+            transaction: transactionName,
+            projectID: decodeScalar(location.query.project),
+            query: location.query,
+          })}
+        />
+      );
+    } else {
+      return <Alert type="warning">{t("You don't have access to this feature")}</Alert>;
+    }
+  };
+
+  render() {
+    const {organization, projects, location} = this.props;
+    const {eventView} = this.state;
+    const transactionName = getTransactionName(location);
+    if (!eventView || transactionName === undefined) {
+      // If there is no transaction name, redirect to the Performance landing page
+      browserHistory.replace({
+        pathname: `/organizations/${organization.slug}/performance/`,
+        query: {
+          ...location.query,
+        },
+      });
+      return null;
+    }
+
+    return (
+      <SentryDocumentTitle title={this.getDocumentTitle()} objSlug={organization.slug}>
+        <Feature
+          features={['measurements']}
+          organization={organization}
+          renderDisabled={this.renderNoAccess}
+        >
+          <GlobalSelectionHeader>
+            <StyledPageContent>
+              <LightWeightNoProjectMessage organization={organization}>
+                <RumContent
+                  location={location}
+                  eventView={eventView!}
```

Comment:
You probably don't need `!` here since you have eventView and transaction name guarded above?

Reply:
Typescript was being weird, but I'll double check if it's needed and remove it if possible.

Reply:
Addressed in 33a7282